src/api/apiService.js

```diff
@@ -2,159 +2,83 @@ import { sessionService } from 'redux-react-native-session';
 import humps from 'humps';
 import Config from 'react-native-config';
 
-const saveSessionHeaders = (headers) => {
-  if (headers.get('access-token')) {
-    const sessionHeaders = {
-      token: headers.get('access-token'),
-      uid: headers.get('uid'),
-      client: headers.get('client')
-    };
-    sessionService.saveSession(sessionHeaders);
-  }
-};
+import handleErrors from 'api/utils/handleErrors';
+import getResponseBody from 'api/utils/getResponseBody';
 
-const handleErrors = response =>
-  new Promise((resolve, reject) => {
-    if (!response) {
-      reject({ message: 'No response returned from fetch' });
-      return;
-    }
-
-    if (response.ok) {
-      saveSessionHeaders(response.headers);
-      resolve(response);
-      return;
-    }
+const ACCESS_TOKEN = 'access-token';
+const APPLICATION_JSON = 'application/json';
+const CONTENT_TYPE = 'Content-Type';
 
-    sessionService.loadSession()
-      .then(() => {
-        if (response.status === 401) {
-          sessionService.deleteSession();
-        }
-      });
+const HTTP_VERB = {
+  GET: 'get',
+  POST: 'post',
+  DELETE: 'delete',
+  PUT: 'put',
+  PATCH: 'patch',
+};
 
-    response.json()
-      .then((json) => {
-        const error = json || { message: response.statusText };
-        reject(error);
-      }).catch(() => reject({ message: 'Response not JSON' }));
-  });
+class Api {
+  static async get(uri, apiUrl = Config.API_URL) {
```

Comment:
you don't need to use `async` if you are not waiting for anything.

Reply:
In my previous comment, I suggested to use `async` with an await for the `loadHeadersAndPerformRequest`. But if you are just returning the promise it's fine without the `async`

Reply:
The use of `await` is implicit there. In fact, if i put it i get an error from the lintern

> error  Redundant use of `await` on a return value

Reply:
My suggestion is to remove the `async`, not to add the `await`. Sorry for the confusion.